lib/cleanupPattern.ts

```diff
@@ -0,0 +1,37 @@
+import { GraphQLNonNull, GraphQLString } from 'graphql';
+
+import { ValidateFunction } from './ValidateDirectiveVisitor';
+import createValidateDirectiveVisitor from './createValidateDirectiveVisitor';
+import createPatternHandler, {
+  PatternDirectiveArgs,
+  defaultArgs,
+} from './pattern-common';
+
+type CleanUpPatternArgs = PatternDirectiveArgs & { replaceWith: string };
+
+const createValidate = ({
+  regexp,
+  flags = null,
+  replaceWith,
+}: CleanUpPatternArgs): ValidateFunction | undefined => {
+  if (!regexp) return undefined;
+  const re = new RegExp(regexp, flags || undefined);
+  return createPatternHandler((value: string): string =>
+    value.replace(re, replaceWith),
+  );
+};
+
+export default createValidateDirectiveVisitor({
+  createValidate,
+  defaultName: 'cleanupPattern',
```

Comment:
nit: Since we can use this directive like `@cleanupPattern(regexp: 'original', replateWith: 'replace')`, the name looks strange to me. Maybe `replacePattern`?

Reply:
@luisfvieirasilva while I see your point, it's in the validation and it's more meant to cleanup stuff. It happens that we allow for custom replacement string (eventually it's `?` instead of the empty string), but providing it as `replacePattern`, to me, will give it the wrong idea (and requests, you'll get reports saying "make it work like `sed`, take `\1` and the likes)

Reply:
i agree with @barbieri